# Literature

## (Amplicon) statistics in general

GUSTA ME Guide to statistical analysis in microbial ecology http://mb3is.megx.net/gustame

Phyloseq https://joey711.github.io/phyloseq/ and especially the FAQs, many great tipps and discussion threads to be found here

## Compositional data analysis

### Websites

Using CoDa for sequence analysis (ALDEx2) https://ggloor.github.io/ 

And a link to a workshop held by G. Gloor about CoDa analysis for sequencing data (Microbiome workshop supplementary) https://github.com/ggloor/CoDa_microbiome_tutorial 

### Papers

The paper that started the whole discussion: Rarefy or not: McMurdie, Paul J., and Susan Holmes. "Waste not, want not: why rarefying microbiome data is inadmissible." PLoS computational biology 10.4 (2014): e1003531.

Best paper with practical tips: Gloor, G. B., Macklaim, J. M., Pawlowsky-Glahn, V., & Egozcue, J. J. (2017). Microbiome datasets are compositional: and this is not optional. Frontiers in microbiology, 8, 2224.

Correlation analysis: Badri, Michelle, Zachary Kurtz, Christian Muller, and Richard Bonneau. 2018. “Normalization Methods for Microbial Abundance Data Strongly Affect Correlation Estimates,” August. https://doi.org/10.1101/406264.

### Books

Microbiome Analysis Methods and Protocols, Beiko et al, Springer
This book mentions state of the art analyses, an example for 18S rRNA gene amplicon workflow, a chapter about data normalization, ...

## Courses

Workshop on Genomics (course, experts, lectures, scripts, examples, ...) http://evomics.org/ 

## Programming in Tidyverse/data analysis

Online book, **buy it on paper**: https://r4ds.had.co.nz/

## Tree of life and GTDB

Hug, Laura A., Brett J. Baker, Karthik Anantharaman, Christopher T. Brown, Alexander J. Probst, Cindy J. Castelle, Cristina N. Butterfield, et al. 2016. ???A New View of the Tree of Life.??? Nature Microbiology, April, 16048. https://doi.org/10.1038/nmicrobiol.2016.48.

Parks, Donovan H., Maria Chuvochina, David W. Waite, Christian Rinke, Adam Skarshewski, Pierre-Alain Chaumeil, and Philip Hugenholtz. 2018. ???A Standardized Bacterial Taxonomy Based on Genome Phylogeny Substantially Revises the Tree of Life. Nature Biotechnology, August. https://doi.org/10.1038/nbt.4229.

Review on Amplicon, metagenomic and metatranscriptomics data analysis: Knight, Rob, et al. "Best practices for analysing microbiomes." Nature Reviews Microbiology (2018): 1.

Primer choice and mock community paper: Parada, A. E., Needham, D. M., & Fuhrman, J. A. (2016). Every base matters: assessing small subunit rRNA primers for marine microbiomes with mock communities, time series and global field samples. Environmental microbiology, 18(5), 1403-1414.

## Live or dead?

Blazewicz, Steven J., Romain L. Barnard, Rebecca A. Daly, and Mary K. Firestone. 2013. Evaluating RRNA as an Indicator of Microbial Activity in Environmental Communities: Limitations and Uses. The ISME Journal 7 (11): 2061-68. https://doi.org/10.1038/ismej.2013.102.

(The below is an attempt of testing activity by mapping RNA reads to DNA amplicon OTU centroid sequences. It turned out not to work so well, so instead, do amplification of RNA and DNA separately so you get amplicons of both.

Thureborn, Petter, Yue O.O. Hu, Andrea Franzetti, Sara Sjoeling, and Daniel Lundin. 2016. Accumulation of DNA in an Anoxic Sediment  RDNA and RRNA Presence of Members of the Microbial Community from Landsort Deep, the Baltic Sea. https://doi.org/10.7287/peerj.preprints.2051v1.)

